src/stages.js

```diff
@@ -0,0 +1,121 @@
+import exec from 'k6/execution'
+
+// parseDuration parses the provided string as an Integer number
+// in millisecond precision
+function parseDuration(str) {
+	let d = 0;
+	let current = '';
+	let isNumber = function(c) {
+		return c >= '0' && c <= '9'
+	}
+
+	for (let i = 0; i < str.length; i++) {
+		if (isNumber(str[i]) || str[i] == '.') {
+			current += str[i]
+		}
+
+		if (isNumber(str[i+1]) || str[i+1] == '.') {
+			continue
+		}
+	
+		let v = parseFloat(current, 10)
+		let next = str[i+1]
+		
+		switch (next) {
+			case 'd':
+				d += v*24*60*60*1000
+				break;
+			case 'h':
+				d += v*60*60*1000
+				break;
+			case 'm':
+				if (i + 2 < str.length && str[i+2] == 's') {
+					d += Math.trunc(v)
+					i++
+				} else {
+					d += v*60*1000
+				}
+				break;
+			case 's':
+				d += v*1000
+				break;
+			default:
+				d += v
+		}
+		i++
+		current = ''
+	}
+	return d 
+}
+
+// getCurrentStageIndex returns the computed index of the running stage.
+function getCurrentStageIndex() {
+	let scenario = exec.test.options.scenarios[exec.scenario.name]
+	if (scenario == null)	{
+		throw new Error(`the exec.test.options object doesn't contain the current scenario ${exec.scenario.name}`)
+	}
+	if (scenario.stages == null) {
+		throw new Error(`only ramping-vus or ramping-arravial-rate supports stages, it is not possible get a stage index on other executors.`)
+	}
+
+	if (scenario.stages.length < 1) {
+		throw new Error(`the current scenario ${scenario.name} doesn't contain any stage`)
+	}
+
+	let sum = 0;
+	let elapsed = new Date() - exec.scenario.startTime
+	for (let i = 0; i < scenario.stages.length; i++) {
+		sum += parseDuration(scenario.stages[i].duration)
+		if (elapsed < sum) {
+			return i
+		}
+	}
+
+	return scenario.stages.length-1
+}
+
+// tagWithCurrentStageIndex adds a tag with a `stage` key
+// and the index of the current running stage as value.
+function tagWithCurrentStageIndex() {
+  exec.vu.tags['stage'] = getCurrentStageIndex()
+}
+
+// tagWithCurrentStageProfile adds a tag with a `stage` key
+// and the profile (ramp-up, steady or ramp-down) computed
+// from the current running stage.
+function tagWithCurrentStageProfile() {
+  //ramp-up when previous.target < current.target
+  //ramp-down when previous.target > current.target
+  //steady when prevuious.target = current.target
+	
+	let getStageProfile = function() {	
+		let currentIndex = getCurrentStageIndex()
+		if (currentIndex < 1) {
+			return 'ramp-up'
+		}
+
+		let stages = exec.test.options.scenarios[exec.scenario.name].stages
+		let current = stages[currentIndex]
+		let previous = stages[currentIndex-1] 
+
+		if (current.target  > previous.target) {
+			return 'ramp-up'
+		}
+		
+		if (previous.target == current.target) {
+			return 'steady'
+		}
+
+		return 'ramp-down'
+	}
+
+	// TODO: should we use another tag key?
+  exec.vu.tags['stage'] = getStageProfile()
```

Comment:
Opinions?

Reply:
My opinion is that at least `tagWithCurrentStageIndex` and `tagWithCurrentStageProfile` should not use the same key. I mean, what if a customer wants to have them both? :thinking: